app/main.py

```diff
@@ -1,34 +1,172 @@
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.decks = self.create_decks()
+        self.is_drowned = is_drowned
+
+    def __len__(self) -> int:
+        return len(self.decks)
+
+    def create_decks(self) -> dict[tuple[int, int], Deck]:
+        decks = {}
+
+        if self.start[0] == self.end[0]:
+            column = self.start[1]
+            while column <= self.end[1]:
+                decks[(self.start[0], column)] = Deck.ALIVE
+                column += 1
+        else:
+            row = self.start[0]
+            while row <= self.end[0]:
+                decks[(row, self.start[1])] = Deck.ALIVE
+                row += 1
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        return decks
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_adjacent_cells(self) -> list[tuple[int, int]]:
+        min_row, min_column = self.start
+        max_row, max_column = self.end
+
+        adjacent_cells = []
+        for row in range(min_row - 1, max_row + 2):
+            for column in range(min_column - 1, max_column + 2):
+                if (row, column) not in self.decks:
+                    if 0 <= row <= 9 and 0 <= column <= 9:
+                        adjacent_cells.append((row, column))
+        return adjacent_cells
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        return self.decks[(row, column)]
+
+    def fire(self, row: int, column: int) -> None:
+        if not self.is_drowned:
+            self.decks[(row, column)] = self.decks[(row, column)].next()
+            if all(status == Deck.HIT for status in self.decks.values()):
+                self.is_drowned = True
+                for key in self.decks:
+                    self.decks[key] = self.decks[key].next()
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(
+            self,
+            ships: list[tuple[tuple[int, int], ...]],
+            title_word: str | None = "РЕСПУБЛІКА"
+    ) -> None:
+        self.title_word = title_word
+        self.fields = Battleship.create_fields(ships)
+        self._validate_field()
+
+    @staticmethod
+    def create_fields(
+            ships: list[tuple[tuple[int, int], ...]]
+    ) -> dict[tuple, Ship]:
+        fields = {}
+        for ship in ships:
+            ship_object = Ship(*ship)
```

Comment:
The `Ship` class constructor expects `start` and `end` as separate arguments, but here it is being unpacked from a tuple. Ensure that the `ships` list contains tuples of two tuples, each representing the start and end coordinates of a ship.